kitsune/sumo/static/sumo/js/switching-devices-wizard-manager.js

```diff
@@ -8,9 +8,113 @@
  */
 export default class SwitchingDevicesWizardManager {
   #formWizard = null;
-  #state = {};
+  #fxaRoot = null;
+  #state = {
+    fxa: {
+      // These UTM parameters are fallback values in the event that we
+      // don't get any passed through to us from the entrypoint to this
+      // SUMO page. These are what will ultimately get passed through the
+      // FxA sign-in flow.
+      utm_source: "support.mozilla.org",
+      utm_campaign: "migration",
+      utm_medium: "mozilla-websites",
+
+      entrypoint: "fx-new-device-sync",
+      entrypoint_experiment: null,
+      entrypoint_variation: null,
+      flow_id: null,
+      flow_begin_time: null,
+      context: null,
+    },
+  };
 
   constructor(formWizard) {
     this.#formWizard = formWizard;
+
+    if (!this.#formWizard.hasAttribute("fxa-root")) {
+      throw new Error(
+        "Expected fxa-root on element passed to " +
+          "SwitchingDevicesWizardManager constructor."
+      );
+    }
+    this.#fxaRoot = this.#formWizard.getAttribute("fxa-root");
+
+    // If the page was loaded with any UTM or entrypoint parameters, we'll want
+    // to pass those along, so now we check for those and add them to our state
+    // if need be. Since this occurs during construction, we don't need to call
+    // #updateState().
+    let searchParams = new URLSearchParams(window.location.search);
+
+    for (let paramName of [
+      "utm_source",
+      "utm_campaign",
+      "utm_medium",
+      "entrypoint",
+      "entrypoint_experiment",
+      "entrypoint_variation",
+    ]) {
+      if (searchParams.has(paramName)) {
+        this.#state.fxa[paramName] = searchParams.get(paramName);
+      }
+    }
+
+    // We need to get some query parameters from the FxA server before we
+    // show the user any kind of form to create or sign-in to an account.
+    // See https://mozilla.github.io/ecosystem-platform/relying-parties/reference/metrics-for-relying-parties#relying-party-hosted-email-form.
+    this.#requestMetricsParams();
```

Comment:
Curious if we need to add error handling in `#requestMetricsParams` - like If we hit some kind of error here do we want/need to prevent the user from even trying to move on to the next step and sign in to FxA?

Reply:
I think we probably want to err on the side of still letting the user go down the happy path. Like, the bad news is that we won't be able to measure the success rate of user's going through this flow in the event that something falls over... but I suspect that measurement problems shouldn't cause us to block the user from signing in.